From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gr1mly4Memes <alagozantoni14@gmail.com>
Date: Mon, 23 Mar 2026 14:35:57 -0700
Subject: [PATCH] test1


diff --git a/src/main/java/ca/spottedleaf/moonrise/common/util/MoonriseCommon.java b/src/main/java/ca/spottedleaf/moonrise/common/util/MoonriseCommon.java
index b7c7de9797d310b68228faf21164ecc701590a59..e03decb8a1297b0a637419351d70fcf0029318f1 100644
--- a/src/main/java/ca/spottedleaf/moonrise/common/util/MoonriseCommon.java
+++ b/src/main/java/ca/spottedleaf/moonrise/common/util/MoonriseCommon.java
@@ -13,7 +13,8 @@ public final class MoonriseCommon {
 
     private static final Logger LOGGER = LogUtils.getClassLogger();
 
-    public static final long WORKER_QUEUE_HOLD_TIME = (long)(20.0e6); // 20ms
+
+    public static final long WORKER_QUEUE_HOLD_TIME = (long)(5.0e6); // 5ms
     public static final BalancedPrioritisedThreadPool WORKER_POOL = new BalancedPrioritisedThreadPool(
         WORKER_QUEUE_HOLD_TIME,
             new Consumer<>() {
@@ -36,11 +37,9 @@ public final class MoonriseCommon {
     public static final BalancedPrioritisedThreadPool.OrderedStreamGroup SERVER_GROUP = MoonriseCommon.WORKER_POOL.createOrderedStreamGroup();
 
     public static void adjustWorkerThreads(final int configWorkerThreads, final int configIoThreads) {
-        int defaultWorkerThreads = OSNuma.getNativeInstance().getTotalCores()  / 2;
+        int defaultWorkerThreads = OSNuma.getNativeInstance().getTotalCores();
         if (defaultWorkerThreads <= 4) {
-            defaultWorkerThreads = defaultWorkerThreads <= 3 ? 1 : 2;
-        } else {
-            defaultWorkerThreads = defaultWorkerThreads / 2;
+            defaultWorkerThreads = defaultWorkerThreads <= 2 ? 2 : 4;
         }
         defaultWorkerThreads = Integer.getInteger(PlatformHooks.get().getBrand() + ".WorkerThreadCount", Integer.valueOf(defaultWorkerThreads));
 
@@ -50,7 +49,7 @@ public final class MoonriseCommon {
             workerThreads = defaultWorkerThreads;
         }
 
-        final int ioThreads = Math.max(1, configIoThreads);
+        final int ioThreads = Math.max(2, configIoThreads <= 0 ? OSNuma.getNativeInstance().getTotalCores() / 2 : configIoThreads);
 
         WORKER_POOL.adjustThreadCount(workerThreads);
         IO_POOL.adjustThreadCount(ioThreads);
@@ -58,7 +57,7 @@ public final class MoonriseCommon {
         LOGGER.info(PlatformHooks.get().getBrand() + " is using " + workerThreads + " worker threads, " + ioThreads + " I/O threads");
     }
 
-    public static final long IO_QUEUE_HOLD_TIME = (long)(25.0e6); // 25ms
+    public static final long IO_QUEUE_HOLD_TIME = (long)(5.0e6); // 5ms
     public static final BalancedPrioritisedThreadPool IO_POOL = new BalancedPrioritisedThreadPool(
         IO_QUEUE_HOLD_TIME,
             new Consumer<>() {
diff --git a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
index cc1906ef871de6a4ff82039d8ddfcd764139f37d..6d0d5c16d44cf52c7e0fccb512372f9af45037ea 100644
--- a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
+++ b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
@@ -41,14 +41,14 @@ public class GlobalConfiguration extends ConfigurationPart {
 
     public class ChunkLoadingBasic extends ConfigurationPart {
         @Comment("The maximum rate in chunks per second that the server will send to any individual player. Set to -1 to disable this limit.")
-        public double playerMaxChunkSendRate = 75.0;
+        public double playerMaxChunkSendRate = 1500.0;
 
         @Comment(
             "The maximum rate at which chunks will load for any individual player. " +
             "Note that this setting also affects chunk generations, since a chunk load is always first issued to test if a" +
             "chunk is already generated. Set to -1 to disable this limit."
         )
-        public double playerMaxChunkLoadRate = 100.0;
+        public double playerMaxChunkLoadRate = 2000.0;
 
         @Comment("The maximum rate at which chunks will generate for any individual player. Set to -1 to disable this limit.")
         public double playerMaxChunkGenerateRate = -1.0;
@@ -67,13 +67,13 @@ public class GlobalConfiguration extends ConfigurationPart {
             "Specifies the maximum amount of concurrent chunk loads that an individual player can have." +
             "Set to 0 to let the server configure it automatically per player, or set it to -1 to disable the limit."
         )
-        public int playerMaxConcurrentChunkLoads = 0;
+        public int playerMaxConcurrentChunkLoads = -1;
 
         @Comment(
             "Specifies the maximum amount of concurrent chunk generations that an individual player can have." +
             "Set to 0 to let the server configure it automatically per player, or set it to -1 to disable the limit."
         )
-        public int playerMaxConcurrentChunkGenerates = 0;
+        public int playerMaxConcurrentChunkGenerates = -1;
     }
     static void set(final GlobalConfiguration instance) {
         GlobalConfiguration.instance = instance;
diff --git a/src/main/java/io/papermc/paper/configuration/WorldConfiguration.java b/src/main/java/io/papermc/paper/configuration/WorldConfiguration.java
index d34e627f8b6711d443553af74113d096b783b36d..89c39b706a09fda442641a4af737fdba9ec57e79 100644
--- a/src/main/java/io/papermc/paper/configuration/WorldConfiguration.java
+++ b/src/main/java/io/papermc/paper/configuration/WorldConfiguration.java
@@ -502,10 +502,10 @@ public class WorldConfiguration extends ConfigurationPart {
 
     public class Chunks extends ConfigurationPart {
         public AutosavePeriod autoSaveInterval = AutosavePeriod.def();
-        public int maxAutoSaveChunksPerTick = 24;
-        public int fixedChunkInhabitedTime = -1;
-        public boolean preventMovingIntoUnloadedChunks = false;
-        public Duration delayChunkUnloadsBy = Duration.of("10s");
+        public int maxAutoSaveChunksPerTick = 200;
+        public int fixedChunkInhabitedTime = 0;
+        public boolean preventMovingIntoUnloadedChunks = true;
+        public Duration delayChunkUnloadsBy = Duration.of("2s");
         public Reference2IntMap<EntityType<?>> entityPerChunkSaveLimit = Util.make(new Reference2IntOpenHashMap<>(BuiltInRegistries.ENTITY_TYPE.size()), map -> {
             map.defaultReturnValue(-1);
             map.put(EntityType.EXPERIENCE_ORB, -1);
